src/fuzz_introspector/analyses/function_call_analyser.py

```diff
@@ -0,0 +1,267 @@
+# Copyright 2022 Fuzz Introspector Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Analysis for function call coverage in the project"""
+
+import logging
+
+from typing import (
+    List,
+    Tuple,
+    Dict
+)
+
+from fuzz_introspector import analysis
+from fuzz_introspector import cfg_load
+from fuzz_introspector import html_helpers
+from fuzz_introspector import utils
+from fuzz_introspector.datatypes import (
+    project_profile,
+    fuzzer_profile,
+    function_profile
+)
+
+logger = logging.getLogger(name=__name__)
+
+
+class Analysis(analysis.AnalysisInterface):
+    def __init__(self) -> None:
+        pass
+
+    @staticmethod
+    def get_name():
+        return "ThirdPartyAPICoverageAnalyser"
+
+    def get_source_file(self, callsite) -> str:
+        src_file = callsite.src_function_source_file
+        if not src_file:
+            parent = callsite.parent_calltree_callsite
+            if parent:
+                src_file = parent.dst_function_source_file
+                src_file = src_file if src_file else ""
+
+        return src_file
+
+    def get_parent_func_name(self, callsite) -> str:
+        func_file = callsite.src_function_source_file
+        if not func_file:
+            parent = callsite.parent_calltree_callsite
+            if parent:
+                func_file = parent.dst_function_name
+                func_file = func_file if func_file else ""
+
+        return func_file
+
+    def add_callsite_record(
+        self,
+        target_func_list: List[str],
+        func_name: str,
+        source_file_list: List[str],
+        callsites: Dict[str, List[str]]
+    ) -> List[str]:
+        exist_list = []
+        if func_name in target_func_list:
+            if func_name in callsites.keys():
+                func_list = callsites[func_name]
+            else:
+                func_list = []
+            for item in source_file_list:
+                if item not in func_list:
+                    func_list.append(item)
+                else:
+                    exist_list.append(item)
+            callsites.update({func_name: func_list})
+
+        return exist_list
+
+    def third_party_func_profile(
```

Comment:
Please provide function-level comments for functions that are non-trival, e.g. https://github.com/ossf/fuzz-introspector/blob/3b265e8f83dbb8dd83c6f7382486b34a7c0bdd2a/src/fuzz_introspector/analyses/optimal_targets.py#L59-L72

Reply:
added